tensorflow/stream_executor/rocm/rocm_blas.cc

```diff
@@ -1621,16 +1622,41 @@ bool ROCMBlas::DoBlasGemm(Stream *stream, blas::Transpose transa,
                       "precondition violation";
     }
   }
-  const Eigen::half alpha_half(alpha);
-  const Eigen::half beta_half(beta);
-  return DoBlasInternal(
+  bool hasXDLOPS = false;
+  int dev = 0;
+  hipGetDevice(&dev);
+  int version = 0;
+  auto status = GpuDriver::GetGpuISAVersion(&version, dev);
+  if (status.ok()) {
+     hasXDLOPS = ((version/100)==9 && (version%100)>=8);
+  }
```

Comment:
two suggestions
1. we need to move away from the number based check (i.e. N >= 908) and towards a string based check ( i.e. "gfxNNN" in {"gfx908", ...} ) to account for future MI HW
2. was wondering whether it would be better to push this check down to the GpuDriver layer (i.e. have a `GpuDriver::HasSupportForXDLOPs()` routine). That way we do not need to copy-paste this logic, should we need to do the same check elsewhere.